cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -125,8 +128,10 @@ func newNodeAnnotator(client clientset.Interface, nodeInformer coreinformers.Nod
 						delete(node.ObjectMeta.Labels, key)
 					}
 
-					for key, value := range labels {
-						node.ObjectMeta.Labels[key] = value
+					err = mergeManagedLabels(node, desiredLabels)
```

Comment:
I am not sure how labels are change upstream but is there any need for mergeManagedLabels to return if any change were made to the annotation to avoid unnecessary write?

Reply:
Is this label-reconciler triggered by controller very often? If frequency is low blind overwrite shall be fine, otherwise I could compare old&new annotation/label to tell whether an update is needed.

Reply:
I haven't watched for Node object updates closely but this controlloop only filters on BootID which is part of the node object's "status".  "status" does include other status type fields like pod counts and tiemstamps.  I wonder if that means there are very frequent updates?  Perhaps you can take a quick look by enabling some log messages at least on test cluster?

Reply:
Tested in sandbox by adding log in the procedure that label-reconcilation is only triggered very few times.

Reply:
BootID only changes on node restart so I do expect that this will be infrequent and that we don't need to over optimize.